tests/dragonfly/replication_test.py

```diff
@@ -368,3 +369,73 @@ def gen_test_data(start, end):
         pipe.get(f"key-{i}")
     vals = await pipe.execute()
     assert all(v is not None for v in vals)
+
+"""
+Test journal rewrites.
+"""
+
+
+@dfly_args({"proactor_threads": 1})
+@pytest.mark.asyncio
+async def test_rewrites(df_local_factory):
+    CLOSE_TIMESTAMP = (int(time.time()) + 100)
+    CLOSE_TIMESTAMP_MS = CLOSE_TIMESTAMP * 1000
+
+    master = df_local_factory.create(port=BASE_PORT)
+    replica = df_local_factory.create(port=BASE_PORT+1)
+
+    master.start()
+    replica.start()
+
+    # Connect clients, connect replica to master
+    c_master = aioredis.Redis(port=master.port)
+    c_replica = aioredis.Redis(port=replica.port)
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    # Make sure journal writer sends its first SELECT command on its only shard
+    await c_master.set("no-select-please", "ok")
+    await asyncio.sleep(0.5)
+
+    # Create monitor and bind utility functions
+    m_replica = c_replica.monitor()
+
+    async def check_rsp(rx):
+        print("waiting on", rx)
+        mcmd = (await m_replica.next_command())['command']
+        print("Got:", mcmd, "Regex", rx)
+        assert re.match(rx, mcmd)
+
+    async def skip_cmd():
+        await check_rsp(r".*")
+
+    async def check(cmd, rx):
+        await c_master.execute_command(cmd)
+        await check_rsp(rx)
+
+    async with m_replica:
+        # CHECK EXPIRE, PEXPIRE, PEXPIRE turn into EXPIREAT
+        await c_master.set("k-exp", "v")
+        await skip_cmd()
+        await check("EXPIRE k-exp 100", r"PEXPIREAT k-exp (.*?)")
```

Comment:
Do you think we can add debug checks to replica that some commands / flows are not executed? So if there is some flow or command we missed we will see this error hopefully

Reply:
Do you mean like making sure `SETEX` never executes on the replica?

Reply:
This is posible also, we could add a check f.e in DbSlice::ExpireParams::Calculate that absolute must be true

Reply:
This is definitely a good idea, but its hard to integrate into dbslice currently. I suggest adding it in the future, because it requires some more efforts